code/modules/mob/inventory.dm

```diff
@@ -209,7 +209,11 @@
 	return FALSE
 
 //Puts the item into our active hand if possible. returns TRUE on success.
-/mob/proc/put_in_active_hand(obj/item/I, forced = FALSE, ignore_animation = TRUE)
+/mob/proc/put_in_active_hand(obj/item/I, forced = FALSE, ignore_animation = TRUE, offered=FALSE)
+	if(!offered) // check if a clicked item is in possession of someone already unless it's offered.
```

Comment:
this seems like a rather snowflake check for something used twice vs a proc used hundreds of places. just add this check to places setting offered = true instead.

Reply:
I wanted a cleaner way, but I couldn't find anything better than adding this snowflake because of how offer code works.